### ISSUE TITLE
Model Marketplace is not updating (Trac #1047)

### ISSUE THREAD

**Issue by butlerpd** [SasView/sasmodel-marketplace#15]
The Model Marketplace is supposed to automatically update whenever a change to a model is pushed to master.  This has not been happening and needs to be investigated.

Migrated from http://trac.sasview.org/ticket/1047
```json
{
    "status": "closed",
    "changetime": "2018-11-14T00:09:06",
    "_ts": "2018-11-14 00:09:06.216260+00:00",
    "description": "The Model Marketplace is supposed to automatically update whenever a change to a model is pushed to master.  This has not been happening and needs to be investigated.",
    "reporter": "butler",
    "cc": "",
    "resolution": "fixed",
    "workpackage": "SasModels Marketplace",
    "time": "2017-12-12T19:00:51",
    "component": "sasmodels Markeplace",
    "summary": "Model Marketplace is not updating",
    "priority": "major",
    "keywords": "",
    "milestone": "SasView 4.2.0",
    "owner": "butler",
    "type": "defect"
}
```

**Comment by pkienzle** [comment]
Trac update at `2018/01/16 17:15:51`: 
* **pkienzle** commented:

The failed marketplace update is causing the travis build of sasmodels master to raise an error.

deploy.sh is raising the following error:

```text
+[ [secure] ]
+ssh-agent -s
+eval SSH_AUTH_SOCK=/tmp/ssh-ZXofcz3OlHPm/agent.4619; export SSH_AUTH_SOCK;
SSH_AGENT_PID=4620; export SSH_AGENT_PID;
echo Agent pid 4620;
+SSH_AUTH_SOCK=/tmp/ssh-ZXofcz3OlHPm/agent.4619
+export SSH_AUTH_SOCK
+SSH_AGENT_PID=4620
+export SSH_AGENT_PID
+echo Agent pid 4620
Agent pid 4620
+chmod 600 .travis/travis_rsa
+ssh-add .travis/travis_rsa
Identity added: .travis/travis_rsa (.travis/travis_rsa)
+git remote add deploy git@danse.chem.utk.edu:/home/git/sasmodels
+git push deploy master
Warning: Permanently added 'danse.chem.utk.edu,160.36.200.68' (RSA) to the list of known hosts.
error: Could not read f728001e783d4f2cad1f52bfa5374c457648c268
To danse.chem.utk.edu:/home/git/sasmodels
 ! [remote rejected] master -> master (shallow update not allowed)
error: failed to push some refs to 'git@danse.chem.utk.edu:/home/git/sasmodels'
Script failed with status 1
```


* **pkienzle** changed priority from "major" to "blocker"

**Comment by butlerpd** [comment]
Trac update at `2018/01/23 14:11:54`: **butler** changed priority from "blocker" to "major"

**Comment by butlerpd** [comment]
Trac update at `2018/02/13 02:49:05`: **butler** changed milestone from "SasView 4.2.0" to "SasView 4.3.0"

**Comment by butlerpd** [comment]
Trac update at `2018/09/08 15:18:09`: 
* **butler** commented:

Tim Snow claims models were never being pushed to the marketplace when changes were made.  He now has a script that could run as a cron job that pulls all models on github master and creates new ones and updates others.

The error noted by Paul Kienzle is separate, and is about deploying market place code change to the server.  Running the script manually from the DANSE server side will cause the above error because Github holds the credentials (RSA key), but if triggered by travis is now working without errors.  Closing this for now.


* **butler** changed resolution from "" to "fixed"
* **butler** changed status from "assigned" to "closed"

**Comment by butlerpd** [comment]
Trac update at `2018/11/14 00:09:06`: 
* **butler** commented:

Moved from 4.2 to 4.3 too quickly apparently -- it was in fact fixed for 4.2 .. though not directly related to the distributed package it was used to finalize 4.2 work on getting models updated in the marketplace.


* **butler** changed milestone from "SasView 4.3.0" to "SasView 4.2.0"
